File: days/day1.go
package days

import (
	"bufio"
	"os"
	"strconv"
)

func Day1() (int, error) {
	f, err := os.Open(".data/day1.txt")
	if err != nil {
		return -1, err
	}

	r := bufio.NewReader(f)

	pos := 50
	count := 0
	for {
		line, err := r.ReadString('\n')
		if err != nil {
			break
		}

		lr := line[0]
		amount, err := strconv.Atoi(line[1 : len(line)-1])
		if err != nil {
			return -1, err
		}
		switch lr {
		case 'L':
			pos -= amount
		case 'R':
			pos += amount
		}

		pos = (pos + 100) % 100

		if pos == 0 {
			count++
		}
	}

	defer f.Close()

	return count, nil
}

File: day1.py
#! /usr/bin/env python3

with open(".data/day1.txt") as f:
  pos = 50
  times = 0

  for line in f:
    lr = line[0]
    number = int(line[1:])

    if lr == 'L':
      pos = (pos - number) % 100
    elif lr == 'R':
      pos = (pos + number) % 100

    if pos == 0:
      times += 1

  print(times)
